lib/Server.js

```diff
@@ -656,35 +645,14 @@ class Server {
 
   createServer() {
     if (this.options.https) {
-      // Only prevent HTTP/2 if http2 is explicitly set to false
-      const isHttp2 = this.options.http2 !== false;
-
-      // `spdy` is effectively unmaintained, and as a consequence of an
-      // implementation that extensively relies on Node’s non-public APIs, broken
-      // on Node 10 and above. In those cases, only https will be used for now.
-      // Once express supports Node's built-in HTTP/2 support, migrating over to
-      // that should be the best way to go.
-      // The relevant issues are:
-      // - https://github.com/nodejs/node/issues/21665
-      // - https://github.com/webpack/webpack-dev-server/issues/1449
-      // - https://github.com/expressjs/express/issues/3388
-      if (semver.gte(process.version, '10.0.0') || !isHttp2) {
-        if (this.options.http2) {
-          // the user explicitly requested http2 but is not getting it because
-          // of the node version.
-          this.log.warn(
-            'HTTP/2 is currently unsupported for Node 10.0.0 and above, but will be supported once Express supports it'
-          );
-        }
-        this.listeningApp = https.createServer(this.options.https, this.app);
-      } else {
-        // The relevant issues are:
-        // https://github.com/spdy-http2/node-spdy/issues/350
-        // https://github.com/webpack/webpack-dev-server/issues/1592
+      if (this.options.http2) {
+        // TODO: we need to replace spdy with http2 which is an internal module
```

Comment:
Ideally, we wanna use http2 which is an internal module but we don't have enough time to implement because the interface isn't the same, so we use spdy for all Node versions because spdy supports 10, 12, and 14.